### PR TITLE
Formatting, Re-enabling the CI, Licensing Info and one Bugfix

### PR DESCRIPTION
**Pull request by davschneller**
A refactoring PR with one tiny bugfix.

* We make the clang formatting style conform to the one of SeisSol. The clang-tidy style is still different however—but can be adjusted when needed.
* The unit tests are re-enabled. Also the tests for `clang-format` and `clang-tidy` are included.
* We conform to the REUSE specification now—i.e. every file has licensing information.
* Fixed a bug, where floating point numbers without an explicit sign after the `d` were not converted to a YAML-compatible float. As a result, now, `0d0` is now parsed correctly to `0e0` which is understood by YAML. Previously, it would have worked only for e.g. `0d+0`.
* Add newlines everywhere.

### CODE REVIEW

**Comment by davschneller** [comment]
@ravil-mobile : is it OK for you to be credited in the copyrights like it is right now? (since you had it under your own Github account until this year)

(edit: got the "OK" on slack)

File: LICENSES/MIT.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+MIT License
+
+Copyright (c) <year> <copyright holders>
```
Comment (vikaskurapati) [inline LICENSES/MIT.txt:3]: Is this intentional? Or do we put the year and SeisSol group here?

File: include/FtyPolicies.hpp [hunk 2]
```diff
@@ -6,26 +10,28 @@
 
 namespace fty {
 struct AsLowercase {
-  std::string apply(const std::string &String) {
+  static auto apply(const std::string& String) -> std::string {
```
Comment (vikaskurapati) [inline include/FtyPolicies.hpp:13]: Why replace the return type with auto and then give it a trailing return type here?

File: include/FtyPolicies.hpp [hunk 3]
```diff
@@ -6,26 +10,28 @@
...
+    std::transform(String.begin(), String.end(), ConvertedString.begin(), [](unsigned char Char) {
+      return std::tolower(Char);
+    });
     return ConvertedString;
   }
 };
 
 struct AsUppercase {
-  std::string apply(const std::string &String) {
+  static auto apply(const std::string& String) -> std::string {
```
Comment (vikaskurapati) [inline include/FtyPolicies.hpp:23]: Why replace the return type with auto and then give it a trailing return type here?

File: include/FtyPolicies.hpp [hunk 4]
```diff
@@ -6,26 +10,28 @@
...
+    std::transform(String.begin(), String.end(), ConvertedString.begin(), [](unsigned char Char) {
+      return std::toupper(Char);
+    });
     return ConvertedString;
   }
 };
 
 struct AsOriginal {
-  std::string apply(const std::string &String) { return String; }
+  static auto apply(const std::string& String) -> std::string { return String; }
```
Comment (vikaskurapati) [inline include/FtyPolicies.hpp:33]: Why replace the return type with auto and then give it a trailing return type here?

File: include/StringProcessor.hpp [hunk 5]
```diff
@@ -1,61 +1,65 @@
...
-public:
+  public:
   /**
    * Checks whether the given line starts with 'Character'
    * @param String tests string
    * @param Character tests character
    * @return true if the string starts with 'Character'. Otherwise, false
    */
-  bool startsWith(const std::string &String, const char Character) {
+  static auto startsWith(const std::string& String, const char Character) -> bool {
```
Comment (vikaskurapati) [inline include/StringProcessor.hpp:20]: Why replace the return type with auto and then give it a trailing return type here?

File: tests/BlockProcessorTest.cpp [hunk 6]
```diff
@@ -1,33 +1,39 @@
...
-#include <string>
+#include <iostream>
+#include <iterator>
+#include <list>
 #include <yaml-cpp/yaml.h>
 
 using namespace fty;
 
-bool isEqualBlocks(const BlockT &Left, const BlockT &Right) {
+auto isEqualBlocks(const BlockT& Left, const BlockT& Right) -> bool {
```
Comment (vikaskurapati) [inline tests/BlockProcessorTest.cpp:17]: Why replace the return type with auto and then give it a trailing return type here?

File: tests/helper.cpp [hunk 7]
```diff
@@ -1,3 +1,11 @@
...
+//
+// SPDX-License-Identifier: MIT
+
 #include "helper.hpp"
+#include "FtyDataTypes.hpp"
+#include <utility>
 
-BlockT make_block(StringsT &Content) { return std::make_pair(Content.begin(), --Content.end()); }
\ No newline at end of file
+auto make_block(StringsT& Content) -> BlockT {
```
Comment (vikaskurapati) [inline tests/helper.cpp:9]: Why replace the return type with auto and then give it a trailing return type here?

**Review by vikaskurapati — APPROVED** [review]

File: LICENSES/MIT.txt (hunk 1 above)
Comment (davschneller) [reply LICENSES/MIT.txt:3]: It is—it's sort of a placeholder that gets the `SPDX-FileCopyrightText` entries put in here in spirit.

The REUSE spec wants a copy of each license to be present; hence it is there.

Cf. here: https://reuse.software/faq/#license-templates

**Pull request merged**